src/tools/file_system.ml

```diff
@@ -161,10 +172,27 @@ let copy_file source target ~overwrite ~throttling =
     begin
       let tmp_file = target ^ ".tmp" in
       unlink tmp_file >>= fun () ->
-      Lwt_io.with_file ~mode:Lwt_io.input source
-        (fun ic ->
-         with_tmp_file tmp_file target
-                       (fun oc -> copy_all ic oc)) >>= fun () ->
+      with_fd
+        source
+        ~flags:[ Unix.O_RDONLY; Unix.O_NONBLOCK; ]
+        ~perm:0
+        (fun fd_in ->
+          with_fd
+            tmp_file
+            ~flags:[ Unix.O_WRONLY; Unix.O_CREAT; Unix.O_TRUNC; Unix.O_NONBLOCK; Unix.O_EXCL; ]
```

Comment:
This is different, compared to the `Lwt_io.with_file` it's replacing (which boils down to:
`[Unix.O_WRONLY; Unix.O_CREAT; Unix.O_TRUNC; Unix.O_NONBLOCK]`. The `O_EXCL` makes it refuse copying
over an existing files and symbolic links